__tests__/createRetryFetch.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 American Express Travel Related Services Company, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+const createRetryFetch = require('../src/createRetryFetch');
+const createTimeoutFetch = require('../src/createTimeoutFetch');
```

Comment:
Sorry for the delay
It might be cleaner delineations of anything breaking (failing tests) if the tests don't include other units? @JAdshead do you have a preference?

That is, would it be easier to test `createRetryFetch` giving it a base fetch whose implementation is controlled:
```javascript
const baseFetch = jest.fn();
const retryFetch = createRetryFetch()(baseFetch);

baseFetch
  .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')))
  .mockImplementationOnce(() => Promise.reject(new Error('test rejection error')))
  .mockImplementationOnce(() => Promise.resolve('Fetch Resolved'))

await expect(retryFetch('/test')).resolves.toBe('Fetch Resolved');
expect(baseFetch).toHaveBeenCalledTimes(3); // may not be needed
```

Reply:
@cbacken thanks for the Pr i like really this.  I agree with Nick, lets keep the tests independent.

Reply:
Done. I removed the other units (timeoutFetch), cleaned up the test cases, and added a few failure tests as well. Thank you both very much for giving time to review this PR.